test/distributed/test_c10d_nccl.py

```diff
@@ -1579,6 +1618,17 @@ def test_default_ddp_comm_hooks_nccl(self):
     def test_fp16_compress_wrapper_nccl(self):
         self._test_fp16_compress_wrapper()
 
+    @requires_nccl()
+    @skip_if_lt_x_gpu(2)
+    def test_hook_then_optimizer_nccl(self):
+        self._test_hook_then_optimizer()
+
+    @requires_nccl()
+    @skip_if_lt_x_gpu(2)
+    def test_hook_then_optimizer_nccl_grad_as_bucket_view(self):
+        self._test_hook_then_optimizer(gradient_as_bucket_view=True)
+
```

Comment:
could we add a test when gradient_as_bucket_view and static_graph is enabled?

Reply:
Added in the most recent version.